### ISSUE TITLE
Fix `no-invalid-position-at-import-rule` false negatives for layers with blocks

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#9025]
### What minimal example or steps are needed to reproduce the bug?

```css
@layer foo {}
@import url('foo.css');
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  rules: {
    'no-invalid-position-at-import-rule': [true]
  }
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAbAhgTxgJwAQDMIJdgBfAHQDskBLAWwAcJsAXXAV2xQAoByQiADowAZxG8AlAG4QAGhD4aKGADk0dOIhgAPdQ2XCxc8BEqKA5ghA6mrXLHxp2KNsCq5c2ZzBHwS7j1xeSggAWhpKADc0FBooUKYRGhYaU1C0FnDGZkyvZV4-AG0WLxgAXQCKSlIpKmNIMxpzADFmOgyrERYMZVjKFmFTC0E6ACsRY1gGCcQ3Sg9yEC6emD6WRb9F9BYfdbkAxeXeiMyGi1CutEooNGwoDdwtjN3FqiqQUiA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "17.1.1",
  "stylelint-config-standard": "40.0.0"
}
```

### What did you expect to happen?

I expected an error as `@layer` rules with blocks aren't allowed before `@import`

### What actually happened?

No error

### Do you have a proposal to fix the bug?

Only ignore `@layer` statements that do not have `nodes`:

```js
!(nodeName === 'layer' && typeof node.nodes === 'undefined') &&
```
